MyLibrary/MyLibraryApp.API/Controllers/CategoriesController.cs

```diff
@@ -0,0 +1,38 @@
+﻿using Microsoft.AspNetCore.Mvc;
+using MyLibraryApp.Application.Services.Interfaces;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Runtime.InteropServices;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace MyLibraryApp.API.Controllers
+{
+    [ApiController]
+    [Route("categories")]
+
+    public class CategoriesController :ControllerBase
+    {
+        private readonly ICategoryService categoryService;
+        public CategoriesController(ICategoryService categoryService)
+        {
+            this.categoryService = categoryService;
+        }
+
+
+        [HttpGet]
+        public async Task<IActionResult> GetAllCategories()
+        {
+            var response = await categoryService.ReadAllCAtegoriesAsync();
+
+            return Ok(response);
+        }
+        [HttpGet]
+        public async Task<IActionResult> GetCategoryById(int id)
+        {
+            var response =await categoryService.FindCategoryByIdAsync(id);
```

Comment:
The naming consistency is quite important. Currently, you have

- GetX
- FindX
- ReadX
All these names are ok but you need to choose one and stick to it in order to follow best practices.